File: modules/algorithms/modules/daa/worst-average-and-best-cases/src/LinearSearch.java
public class LinearSearch {

  static int search(int arr[], int arrSize, int element)
  {
    int i;
    for(i = 0; i < arrSize; i++)
      if(arr[i] == element)
        return i;
    return -1;
  }

  public static void main(String[] args)
  {
    int arr[] = {1, 10, 30, 15};
    int element = 30;
    int arrSize = arr.length;

    System.out.printf("%d is present at index %d", element, search(arr, arrSize, element));
  }
}

File: modules/algorithms-and-ds/modules/design-and-analysis-of-algorithms/src/linear_search.cpp
#include <iostream>
using namespace std;

// Linearly search x in arr[].
// If x is present then return the index,
// otherwise return -1
int search(int arr[], int n, int x)
{
    int i;
    for (i = 0; i < n; i++) {
        if (arr[i] == x)
            return i;
    }
    return -1;
}


// Driver's Code
int main()
{
    int arr[] = { 1, 10, 30, 15 };
    int x = 30;
    int n = sizeof(arr) / sizeof(arr[0]);

    // Function call
    cout << x << " is present at index " << search(arr, n, x);

    return 0;
}
